packages/preset-icons/src/core.ts

```diff
@@ -51,9 +51,9 @@ export function createPresetIcons(lookupIconLoader: (options: IconsOptions) => P
           await customizations.iconCustomizer?.(collection, icon, props)
```

Comment:
revert `scale` changes and set this call before `iconCustomizer` call:
```ts
customizations.customize?.(props as unknown as any)
```

Reply:
should be (will work with iconify icons and custom svg icons):
```ts
customizations.customize?.(props as unknown as any)
await customizations.iconCustomizer?.(collection, icon, props)
```

Reply:
running playground with previous changes:

![imagen](https://github.com/user-attachments/assets/8d70a8e0-25eb-4673-81df-e6dcaf035938)

Reply:
@cyberalien we need to review iconify icon loader

Reply:
> should be (will work with iconify icons and custom svg icons):
> 
> ```ts
> customizations.customize?.(props as unknown as any)
> await customizations.iconCustomizer?.(collection, icon, props)
> ```

[check1](https://github.com/iconify/iconify/blob/e60da173c6a25a2cce7eb547519d3093ade82777/packages/utils/src/loader/modern.ts#L39): the **w/h** retruned from `customize()` are not passed to `iconCustomizer()`, which is not like what docs [saying](https://unocss.dev/presets/icons#icon-customizations) - "_apply iconCustomizer with customize customizations, if provided_.". (This should be intentional behaviour? Im not sure.)

[check2](https://github.com/iconify/iconify/blob/e60da173c6a25a2cce7eb547519d3093ade82777/packages/utils/src/loader/modern.ts#L59): And then, the explicit `scale = 1` take precedence over `customize()` to w/h.

So I think it's fine to just drop the destructured default scale value, and the `customize()` should work as expected.

Reply:
Check vite-vue3 example, will not work

Reply:
> Check vite-vue3 example, will not work

Sorry, I can't get your meaning. I have checked this example and it looks fine. What's wrong with it? 

The shorcuts takes precedence as expected.

![image](https://github.com/user-attachments/assets/02b6a444-2d10-47e2-acf4-6d9b38ebe58f)

![image](https://github.com/user-attachments/assets/269445c9-5c37-4685-a8a3-a5b02e06fecc)

Reply:
Sorry, I have to go to bed now. Don't wait for my reply. I appreciate whatever u will do for this issue.

Reply:
The page should have 2 black svg circles,  check the App.vue

Reply:
Inspect the pages, both there but missing width/height in css styled (0px)

Reply:
> @cyberalien we need to review iconify icon loader

Sure, but I can't do that right now. Should be back to computer in 3 days.

Reply:
Checked it and I'm very confused. Not sure why width/height doesn't work or how this pr fixes it.

Reply:
> @cyberalien we need to review iconify icon loader

Issue is not in loader. I've added unit tests to test for this specific issue, loader returns correct icon with customised width and height.

See https://github.com/iconify/iconify/commit/84b4fab85f9817eaf91af74f37de2c6d009264eb

Renders `<svg viewBox="0 0 48 48" width="2em" height="1em" ><defs><path id="flatColorIconsUpRight0" fill="#3F51B5" d="M44 19L30 30.7V7.3z"/><path id="flatColorIconsUpRight1" fill="#3F51B5" d="M6 27v13h8V27c0-2.2 1.8-4 4-4h17v-8H18c-6.6 0-12 5.4-12 12"/></defs><use href="#flatColorIconsUpRight0"/><use href="#flatColorIconsUpRight1"/><use href="#flatColorIconsUpRight0"/><use href="#flatColorIconsUpRight1"/></svg>`

Reply:
Actually, there is an error in loader. It happens when `scale` option is set. It overrides custom value.

Fixed in https://github.com/iconify/iconify/commit/439a862a1a0f0679038a0272bf06a9f277a50756

New behavior scales custom values instead of just returning `${scale}em`

Reply:
Published `@iconify/utils` version `2.1.26` with fix.

Additional changes to `customize` callback:

It now has 2 more parameters:
- Icon data in `IconifyIcon` format, which is a mutable object. Can be used to get width/height ratio if you want to have different width and height. Because this is a mutable object, it can also be used to completely change icon content (change colors in icon with hardcoded colors, change animation duration for animated icons, add custom stuff and so on).
- Icon name in `${collection}:${name}` format, so you can check if you are customising correct icon.

Returning modified `props` is no longer required (but still there for compatibility). It is a mutable object, so callback can just change it. This is done because new icon data parameter is also a mutable object, so wouldn't make much sense to return props if you are modifying only icon data. It makes more sense to just change objects and not return anything.

Reply:
Re-fixed in `2.1.28` (and I skipped a version by accident when publishing update, but it doesn't matter).

Option `scale` was broken for non-square icons, but there were no non-square icons in tests to play with, so issue was unnoticed and all icons were square.

In old version when option `scale` was not set, non-square icons were rendered correctly as non-square. If `scale` was set, non-square icons became square. That also affected wide icons by making them shorter than needed.

Now it is fixed, non-square icons have correct aspect ratio.

Reply:
Added pull request that updates utils and updates unit tests: https://github.com/unocss/unocss/pull/4023

Should also fix this issue.